### PR TITLE
Small improvements to Matrix mobject

### PR DESCRIPTION
**Pull request by PhilippImhof**
easy customization of the left and right bracket of a matrix
- added attributes `left_bracket` and `right_bracket` to `Matrix` class, default is to use old hard-coded brackets
- changed `add_brackets()` to accept parameters specifying left and right bracket to use
- changed `__init__()` in order to call `add_brackets` with the newly introduced parameters

set row colors to a matrix mobject, similar function exists for columns
- added method `get_rows()` similar to `get_columns()`, returning the rows of a matrix as a `VGroup`
- added method `set_row_colors()` similiar to `set_column_colors()`

### CODE REVIEW

**Comment by PhilippImhof** [comment]
This PR has been sent to 3b1b's repository a few months ago, but never got merged as (probably) Grant did not consider this useful to his own projects. It might be different in a community maintained version of manim that is used by many people in different situations.

See https://github.com/3b1b/manim/pull/745 for discussion

**Comment by PhilippImhof** [comment]
This PR has been sent to 3b1b's repository a few months ago, but never got merged as (probably) Grant did not consider this useful to his own projects. It might be different in a community maintained version of manim that is used by many people in different situations.

See https://github.com/3b1b/manim/pull/765 for original pull request.

**Comment by leotrs** [comment]
LGTM. Thank you for your contribution!

**Comment by leotrs** [comment]
BTW: in this repo, at least two members with write access have to leave an approving code review. I just left mine. We need one more so please be patient!

**Comment by PhilippImhof** [comment]
Thank you Leo. I have been waiting in vain for quite a few months for the PR to get merged into 3b1b's manim repository, so I think I can easily wait for a second approval vote on this one :)

**Comment by PhilippImhof** [comment]
Thank you @Aathish04. I will edit the PR text accordingly.

File: manimlib/animation/transform.py [hunk 1]
```diff
@@ -152,7 +152,7 @@ def __init__(self, method, *args, **kwargs):
         method is a method of Mobject, *args are arguments for
         that method.  Key word arguments should be passed in
         as the last arg, as a dict, since **kwargs is for
-        configuration of the transform itslef
+        configuration of the transform itself
```
Comment (PgBiel) [inline manimlib/animation/transform.py:155]: (As I stated elsewhere) Should probably be in a separate general cleanup PR (for organization purposes)

File: manimlib/mobject/matrix.py [hunk 2]
```diff
@@ -100,8 +102,8 @@ def organize_mob_matrix(self, matrix):
                 )
         return self
 
-    def add_brackets(self):
-        bracket_pair = TexMobject("\\big[", "\\big]")
+    def add_brackets(self, left, right):
```
Comment (PgBiel) [inline manimlib/mobject/matrix.py:105]: Should probably default them to `r"\big[", r"\big]"`, jic (also would make this a non-breaking change)

File: manimlib/mobject/svg/brace.py [hunk 3]
```diff
@@ -9,7 +9,7 @@
 from manimlib.mobject.types.vectorized_mobject import VMobject
 from manimlib.utils.config_ops import digest_config
 from manimlib.utils.space_ops import get_norm
-
+import copy
```
Comment (PgBiel) [inline manimlib/mobject/svg/brace.py:12]: (As I stated elsewhere) Should probably be in a separate general cleanup PR (for organization purposes)

**Comment by PgBiel** [comment]
Was this tested?

File: manimlib/mobject/matrix.py (hunk 2 above)
Comment (leotrs) [reply manimlib/mobject/matrix.py:105]: Agreed

File: manimlib/animation/transform.py (hunk 1 above)
Comment (leotrs) [reply manimlib/animation/transform.py:155]: I'm fine with doing small typo fixes while in other PRs. If this contained dozens of doc/typo fixes then I'd ask for separate PRs.

File: manimlib/mobject/matrix.py [hunk 4]
```diff
@@ -125,6 +127,18 @@ def set_column_colors(self, *colors):
             column.set_color(color)
         return self
 
+    def get_rows(self):
```
Comment (PgBiel) [inline manimlib/mobject/matrix.py:130]: Documentation

File: manimlib/mobject/matrix.py [hunk 5]
```diff
@@ -125,6 +127,18 @@ def set_column_colors(self, *colors):
...
+        return VGroup(*[
+            VGroup(*self.mob_matrix[i, :])
+            for i in range(self.mob_matrix.shape[1])
+        ])
+
+    def set_row_colors(self, *colors):
```
Comment (PgBiel) [inline manimlib/mobject/matrix.py:136]: Documentation

File: manimlib/animation/transform.py (hunk 1 above)
Comment (PhilippImhof) [reply manimlib/animation/transform.py:155]: Actually, I do not know why this change is in my PR as it is not my commit. I pulled from master and created my branch for the matrix changes.

File: manimlib/mobject/matrix.py (hunk 2 above)
Comment (PhilippImhof) [reply manimlib/mobject/matrix.py:105]: Fine with my. Sorry for asking, but it seems to me that your suggested change does not use `r"\big["` and `r"\big]"` contrary to what you wrote in your comment. What do we do?

File: manimlib/mobject/svg/brace.py (hunk 3 above)
Comment (PhilippImhof) [reply manimlib/mobject/svg/brace.py:12]: Same as above: This change should not be in my PR, as it has nothing to do with my changes.

**Comment by PhilippImhof** [comment]
@PgBiel Yes, of course, the code was tested. I have used it in several of my own animations.

I will add documentation as requested. Please understand that when I first wrote this, it was meant for 3b1b's repo and therefore follows Grant's verbosity....

File: manimlib/mobject/matrix.py (hunk 2 above)
Comment (safinsingh) [reply manimlib/mobject/matrix.py:105]: That's because he used raw strings in his comment but non-raw strings in the edit; they do the same thing.

Comment (PhilippImhof) [reply manimlib/mobject/matrix.py:105]: @safinsingh Thanks, so I misunderstood. I thought his suggestion was mainly about using raw strings :)

File: manimlib/mobject/svg/brace.py (hunk 3 above)
Comment (PgBiel) [reply manimlib/mobject/svg/brace.py:12]: @PhillipImhof I added a suggestion here as well

**Comment by PgBiel** [comment]
> @PgBiel Yes, of course, the code was tested. I have used it in several of my own animations.
Awesome! We should add Travis tests in the future as well

> I will add documentation as requested. Please understand that when I first wrote this, it was meant for 3b1b's repo and therefore follows Grant's verbosity....
Alright fam, I'll be waiting!

**Comment by PhilippImhof** [comment]
I added short documentation and (hopefully) removed rogue commits by reverting them. However, I do not quite understand why the branch now has "conflicts that must be resolved" whereas there were no conflicts yesterday.

**Comment by PhilippImhof** [comment]
@leotrs I did a `git rebase -i origin/master` and removed all rogue commits. I suppose the problem comes from (me misusing) GitHub Desktop. Instead of just pushing my changes, it magically pulled other commits and added them to the list. One never stops learning.

**Comment by leotrs** [comment]
@PhilippImhof the conflicts are because there's a ton of activity right now on the repo and other PRs have been berged before yours. Not your fault :)

**Comment by leotrs** [comment]
Now we have to wait for @PgBiel's pending review

**Comment by PhilippImhof** [comment]
@leotrs Thanks. I somehow thought that GitHub would keep track of these changes, but in hindsight, that seems to be a stupid misconception as it would quickly become way too complicated.

File: manimlib/mobject/matrix.py (hunk 2 above)
Comment (PgBiel) [reply manimlib/mobject/matrix.py]: yeah sorry 
I prefer raw strings but I wanted to be consistent with the already-existing code

**Comment by leotrs** [comment]
@PhilippImhof and/or @PgBiel would you be so kind as to resolve the conflicts and then merge this? Thanks :)

**Comment by PhilippImhof** [comment]
@leotrs I would gladly do so, but I do not have the necessary permissions.

**Comment by PgBiel** [comment]
> @PhilippImhof and/or @PgBiel would you be so kind as to resolve the conflicts and then merge this? Thanks :)

I'll take a look in a bit

**Pull request merged**